File: 2089-maximum-matrix-sum/solution.cpp
class Solution {
public:
    // #define ll long long ;
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        long long sum=0;
        int n=matrix.size(), cnt=0, mini=INT_MAX;
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                if(matrix[i][j]<0) cnt++;
                mini=min(abs(matrix[i][j]), mini);
                sum+=abs(matrix[i][j]);
            }
        }
        if(!(cnt&1)) return sum;
        return (long long)(sum-2*mini);
    }
};

File: 2089-maximum-matrix-sum/solution.java
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        // find the least abs value k

        // if no of negatives is odd,
        // ans = sum of absolutes - 2 * k

        // else 
        // ans = sum of absolutes

        long sum = 0;
        int mini = (int)1e9, c = 0;
        for(int row [] : matrix){
            for(int x: row){
                if (x<0){
                    c++;
                }
                mini=(int)Math.min(mini,(int)Math.abs(x));
                sum+=(int)Math.abs(x);
            }
        }  

        if ((c&1)==1) return sum - 2 * mini;
        return sum;
    }
}
